Mathlib/Order/Partition/Basic.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2025 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.SetLike.Basic
+import Mathlib.Order.SupIndep
+
+/-!
+# Partitions
+
+A `Partition` of an element `a` in a complete lattice is an independent family of nontrivial
+elements whose supremum is `a`.
+
+An important special case is where `s : Set α`, where a `Partition s` corresponds to a partition
+of the elements of `s` into a family of nonempty sets.
+This is equivalent to a transitive and symmetric binary relation `r : α → α → Prop`
+where `s` is the set of all `x` for which `r x x`.
+
+## Main definitions
+
+* For `[CompleteLattice α]` and `s : α`, a `Set.Partition s` is an independent collection of
+  nontrivial elements whose supremum is `s`.
+
+-/
+variable {α : Type*} {s x y z : α}
+
+open Set
+
+/-- A `Partition` of an element `s` of a `CompleteLattice` is a collection of
+independent nontrivial elements whose supremum is `s`.  -/
+structure Partition [CompleteLattice α] (s : α) where
+  /-- The collection of parts-/
+  parts : Set α
+  /-- The parts are `sSupIndep`-/
+  indep : sSupIndep parts
+  /-- The bottom element is not a part-/
+  bot_not_mem : ⊥ ∉ parts
+  /-- The supremum of all parts is `s`-/
+  sSup_parts_eq : sSup parts = s
+
+namespace Partition
+
+section Basic
+
+variable [CompleteLattice α] {P : Partition s}
+
+instance {α : Type*} [CompleteLattice α] {s : α} : SetLike (Partition s) α where
+  coe := Partition.parts
+  coe_injective' p p' h := by cases p; cases p'; simpa using h
+
+@[simp] lemma coe_parts : (P.parts : Set α) = P := rfl
+
+@[ext] lemma ext {P Q : Partition s} (hP : ∀ x, x ∈ P ↔ x ∈ Q) : P = Q :=
+  SetLike.ext hP
+
+lemma disjoint (hx : x ∈ P) (hy : y ∈ P) (hxy : x ≠ y) :
+    Disjoint x y :=
+  P.indep.pairwiseDisjoint hx hy hxy
+
+lemma pairwiseDisjoint : Set.PairwiseDisjoint (P : Set α) id :=
+  fun _ hx _ hy hxy ↦ P.disjoint hx hy hxy
```

Comment:
```suggestion
  P.indep.pairwiseDisjoint
```
maybe? (untested, but from the way this and the previous lemma are proved, it looks like it might work)